### ISSUE TITLE
Het zoeken naar een opleidingsverantwoordelijke duurt vrij lang

### ISSUE THREAD

**Issue by KWillemse** [Nelos-VZW/dives-2#461]
**Beschrijf de bug**
Als je een nieuwe opleiding start dan staat onze duikschoolleider automatisch ingevuld, maar hij is in onze club voor geen enkele opleiding de OPLV.  Ik moet dus telkens een andere OPLV kiezen en dat duurt vrij lang vooraleer die gevonden is.
Concreet, ik type 'Erika' in de lijst en dat duurt 7 seconden vooraleer die is gevonden. Dat lijkt weinig maar ik moet dit najaar 20 nieuwe opleiding starten dus dan begint dat te tellen...
Als ik een nieuwe duik activiteit start, dan duurt dat zoeken max 1 seconde. 

**Verwachte resultaat**
Ik vermoed dat het zoeken naar een OPLV kan gelimiteerd worden naar de personen in je eigen club. Je zou dit automatisch kunnen limiteren, of je zou een 'switch kunnen voorzien' zoals op verschillende andere plaatsen waarbij de default natuurlijk je eigen club is.

**Screenshots**
![image](https://github.com/user-attachments/assets/85bedf48-5c1c-4b76-b6d0-6901c0a1b016)

**Comment by KWillemse** [comment]
See case #410

**Comment by KWillemse** [comment]
Inderdaad, ik had nochthans gezocht in de database want ik kon me herinneren dat ik dat al eens gelogd had. Niet goed genoeg gezocht dus.
Wel even meegeven dat het ondertussen al 7 seconden duurt per opleiding, dus het wordt trager en trager :-(

**Comment by ndsmyter** [comment]
Ik heb geprobeerd om de query te versnellen, maar er zijn tegenwoordig wel wat meer controles die moeten gebeuren. Het is nu wel al iets sneller dan voorheen.

Verder heb ik eens geprobeerd om dezelfde component te gebruiken als we op andere plaatsen gebruiken voor het zoeken naar buddy's. Waarbij je dus de club kan selecteren (of niet). Maar dat ziet er totaal niet gebruiksvriendelijk uit als je meerdere opleidingen selecteert. Ik heb dat dus teruggedraaid, want dat gaat het niet worden.

**Comment by KWillemse** [comment]
Een ideetje om niet de 'club te selecteren' maar de selectie te gebruiken om enkel te zoeken binnen de clubleden zoals bij het loggen van een activiteit.
Bijvoorbeeld:
![image](https://github.com/user-attachments/assets/7a18d538-c660-4591-9068-c02bcff366aa)

**Comment by ndsmyter** [comment]
Hey @KWillemse ,
Dat is wat ik effectief geprobeerd heb op die manier. Als het allemaal wat meer onder elkaar komt te staan bijvoorbeeld op een GSM, dan ziet dat er heel lelijk uit met al die checkboxen.

Daarnaast werkt onze huidige component niet met de user binnenin, maar moet die gebruikt worden om die user elders op te slaan nadat je die hebt gevonden.

Misschien niet zo duidelijke uitleg, maar betekent gewoon dat de huidige component niet bruikbaar is voor in deze view. Maar dat het voor die paar seconden volgens mij ook nog niet de moeite is om een nieuwe component te maken die quasi identiek is als een andere. Dat is namelijk ook weer een extra component die te onderhouden is.

Dus op dit moment ga ik het alleen proberen met de paar optimalisaties aan de query die ik heb gedaan.

**Comment by KWillemse** [comment]
Voor de user interface zou je kunnen kiezen voor een globale switch die werkt op alle zoekvelden:
![image](https://github.com/user-attachments/assets/9821d41d-d470-4313-9744-f343bf86d701)
Maar of dat deze keuze het achterliggend eenvoudiger op maakt om te implementeren dat weet ik niet.

Ik begrijp het verschil in componenten en ook de maintenance lijkt me een goede reden om het niet te doen. Gewoon spijtig dat snelle filters bij het loggen van activiteiten (filter op eigen leden leden), club beheer (filter op actieve leden), en de kandidaten lijst (filter op favorieten en filter op club zie afbeelding) wel heel vlot gaat.
![image](https://github.com/user-attachments/assets/4e5de1ea-aaa6-49ed-8aaf-3b34aaf0b6ed)

**Comment by KWillemse** [comment]
Ik krijg nu een heel vreemd gedrag bij het zoeken van een nieuwe OPLV. In eerste instantie vind hij de instructeur niet, na wat proberen lukt het wel.
10 nieuwe kandidaten, dus bij gevoegde actie heb ik 10 keer moeten uitvoeren.
Dropbox link bevat een video.
https://www.dropbox.com/scl/fi/fmb854mqgl5avp52xcghs/C.V.D.-Ledenadministratie-Google-Chrome-2025-03-25-13-50-08.mp4?rlkey=qyk4vg7x3x10aqoro10g9rvvl&dl=0

**Comment by ndsmyter** [comment]
Het laatste probleem kan ik niet reproduceren. Mogelijks gaat het om een tijdelijk issue of iets dat ondertussen al opgelost is in de versie waar ik nu mee aan het testen ben.

**Comment by KWillemse** [comment]
Ik heb het probleem nog altijd met de live versie. Ik hoop echt dat deze case deftig opgelost kan geraken tegen het nieuwe duikschooljaar. Ik ben secretaris van een heel grote club, ik moet dus veel administratie werk verrichten. Er worden allerhande software systemen voorzien om de administratie gemakkelijker te maken, bv DIVES, Twizzit, ... en dat is inderdaad het geval voor het grootste deel van de gebruikers.
Dat er prioriteiten worden gezet aan de hand van het aantal gebruikers en de belangrijkheid is de meest logische zaak en ik ken dit maar al te goed door mijn 25 jaar ervaring in de software development business als software tester.
Maar als de software ondertussen al een aantal jaren in gebruik is het tijd om u te focussen op improvements en nieuwe functionaliteit voor de andere gebruikers.
Ik zie ook dat jullie veel werk op de plank hebben vanuit NELOS om verbetering aan te brengen of nieuwe onderdelen toe te voegen maar ik zie ook dat de 'gebruiksvriendelijkheid' voor de ledenadministrators achterwege blijft. 
Een paar seconden extra is voor de modale ledenadministrator geen probleem, maar voor de grotere clubs is dit niet werkbaar en brengt daardoor geen verbetering. 
Ik hoop dat er in de planning voor de ook eens een onderdeel 'ledenadministrator verbeteringen' kan worden opgenomen.
